### ISSUE TITLE
Regression in Fable.Remoting.Client 7.9 causing XMLHttpRequest errors to be ignored

### ISSUE THREAD

**Issue by bklop** [Zaid-Ajaj/Fable.Remoting#249]
Commit 4cd23641182ca82e27dfafa25dc827295840b28b added support for cancellation, but also checks for `xhr.status <> 0` which causes it to swallow any `XMLHttpRequest` errors.

In practice this means that client code such as this (using Elmish here as an example)

```
Cmd.OfAsync.either
    MyProxy.someApi.doSomething
    ()
    SomethingSuccess
    SomethingError
```
will never dispatch a SomethingError msg in case of a network error (for example, a lost connection). This is breaking application functionality for us and we were forced to rollback to Fable.Remoting.Client 7.8 for now :).

**Comment by Zaid-Ajaj** [comment]
Hi there @bklop thanks for filing the issue, this is interesting because I thought [this test](https://github.com/Zaid-Ajaj/Fable.Remoting/blob/master/ClientV2Tests/src/App.fs#L568) would make sure we are throwing proper errors 🤔 so is this only when status code = 0? (i.e. cors/connectivity error)

**Comment by bklop** [comment]
Yeah exactly. I'm not sure why `xhr.status <> 0` was added in that commit, shouldn't it be sufficient to check for `ReadyState.Done && not token.IsCancellationRequested`?

**Comment by Zaid-Ajaj** [comment]
Done! The fix should be available in v7.11 🚀
